app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -673,6 +673,8 @@ def build_toolbar_hide_button(id)
         return true unless @record.is_refreshable?
       when "host_scan"
         return true unless @record.is_scannable?
+      when "host_toggle_maintenance"
+        return true unless (@record.respond_to?(:set_node_maintenance) && @record.respond_to?(:unset_node_maintenance))
```

Comment:
The above two lines should be implemented not here in `toolbar_builder.rb` but in a button class of its own.

Similar to what's already implemented for other buttons under `app/helpers/application_helper/button/`

In this case, the button class would be something like:

```ruby
class ApplicationHelper::Button::HostToggleMaintenance < ApplicationHelper::Button::Basic
  def disabled?
    @record.is_available?(...) && @record.is_available?(...)
  end
end
```

Yes, it's confusing that some of the "disabled button" functionality is implemented in `toolbar_builder.rb` and
some in its own button class, but we want to move away from having this in `toolbar_builder.rb` and have
this functionality in separate button classes (read application pluggability). So for any new code, a button class
should be used.